### PR TITLE
Adds autodocs

### LINKED ISSUE #27: Add autodocs
**Is your feature request related to a problem? Please describe.**
As we start to write code, it will be useful to have a mechanism that automatically creates some documentation about each class and function, so that we don't have to constantly update the readme with function definitions.

**Describe the solution you'd like**
Explore and implement a solution which generates documentation based on docstrings.

### PR DESCRIPTION
**Pull request by daremiyo**
## Changes

- Adds autodocs
- Adds autodoc directives for data_helpers
- Sets up autodocs config to organize directives by source
- Sets up autodocs config to mock import from fuzzywuzzy

## Questions

- Is by source the desired arrangement for autodoc directives?
- Mocked import from fuzzwuzzy because the import blocked sphinx from generating the html site. Error said that sphinx could not find module for fuzzwuzzy. Is mock import the best strategy for this scenario? 

Fixes #27

### CODE REVIEW

File: .docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# The full version, including alpha/beta/rc tags
+release = '1.0'
+
+
+# -- General configuration ---------------------------------------------------
+
+# Add any Sphinx extension module names here, as strings. They can be
+# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
+# ones.
+extensions = [ 'sphinx.ext.autodoc', 'sphinx.ext.coverage', 'sphinx.ext.napoleon'
```
Comment (p-galligan) [inline .docs/conf.py]: You should remove the space after `[` here for linting.

File: .docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+]
```
Comment (p-galligan) [inline .docs/conf.py]: Fix indenting here to line up `]` with opening bracket to pass linting.

**Review by p-galligan — CHANGES_REQUESTED** [review]
I think a couple of changes will get this to pass linting.

File: .docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+#
+import os
+import sys
+sys.path.insert(0, os.path.abspath('..'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'rac_aspace'
+copyright = '2020, Hillel Arnold, Amy Berish, Patrick Galligan, Bonnie Gordon, Katie Martin, Hannah Sistrunk, Darren Young'
```
Comment (helrond) [inline .docs/conf.py]:
```suggestion
copyright = '2020, Rockefeller Archive Center'
```

File: .docs/conf.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+author = 'Hillel Arnold, Amy Berish, Patrick Galligan, Bonnie Gordon, Katie Martin, Hannah Sistrunk, Darren Young'
```
Comment (helrond) [inline .docs/conf.py]:
```suggestion
author = 'Rockefeller Archive Center staff'
```

**Review by helrond — CHANGES_REQUESTED** [review]
This is great @daremiyo, just a few small changes to align with past practice around corporate authorship!

File: .docs/conf.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+release = '1.0'
+
+
+# -- General configuration ---------------------------------------------------
+
+# Add any Sphinx extension module names here, as strings. They can be
+# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
+# ones.
+extensions = ['sphinx.ext.autodoc', 'sphinx.ext.coverage', 'sphinx.ext.napoleon'
+             ]
```
Comment (p-galligan) [inline .docs/conf.py]:
```suggestion
              ]
```

**Comment by p-galligan** [comment]
Looks good to me now! Looks like it's just failing on 3.4 and 3.5 which could be fixed by changing some info in the requirements file. You'd have to downgrade pre-commit to an earlier version. Just re-request review whenever you're ready.

**Comment by HaSistrunk** [comment]
I'm wondering if it's ok to leave the pre-commit version as is in the PR since PR #66 will fix this issue?

**Comment by p-galligan** [comment]
> I'm wondering if it's ok to leave the pre-commit version as is in the PR since PR #66 will fix this issue?

I personally think that's fine. It's passing most tests now.

**Comment by helrond** [comment]
I think there's an order of operations thing here - we're going to have a lot of merge conflicts in the requirements file, so I think we may need to talk about what gets merged first, and how important it is for us to merge only working code. I think there are two paths:
- Merge #66 first, and then update each of the PRs to use those multiple requirements files. Downside is a lot of fixing PRs, upside is that we don't merge non-working code.
- Merge everything BUT #66 first, regardless of whether it's passing CI, and then merge #66, which should allow everything to pass. Downside is there might be additional bugs exposed which would have to be fixed, upside is fewer PR fixes.

Now that I'm writing this out, I think I'd prefer the first option - it keeps problems isolated and diagnosable.

**Review by p-galligan — APPROVED** [review]

**Review by helrond — APPROVED** [review]

**Pull request merged**